src/services/socket.tsx

```diff
@@ -3,40 +3,87 @@
 import { io } from 'socket.io-client'
 import { useEffect, useState } from 'react'
 
-type ListenerRes = {
-	messageServer: string
-	roomIdServer: string
-	userNameServer: string
+export type ListenerRes = {
+	message: string
+	// roomId: string
+	userName: string
+	timeStamp: string
 }
 
-export interface Event extends ListenerRes {
+type AllUsersPointsRes = {
+	message: ListenerRes[]
+	// roomId: string
+	userName: string
 	timeStamp: string
 }
 
 const socket = io()
 
 export function socketRoomEmitter(
 	emitterName: string,
-	message: string,
-	roomId: string,
+	message: string | ListenerRes[],
 	userName: string,
+	timeStamp: string,
```

Comment:
todo: Have you noticed that you do timeStamp over and over again across the app? This is what often called "meta" data about an event. That could be done inside this function directly:

nit: It's common to write timestamp, not timeStamp.

```js
  const timestamp = Date.now().toString()
	socket.emit(emitterName, message, userName, timestamp, roomId)
```

Reply:
I had already noticed that and fixed it. It will be in the next commit.